src/features/generic_events/aggregate/user-actions/user-actions-aggregator.js

```diff
@@ -0,0 +1,95 @@
+import { generateSelectorPath } from '../../../../common/dom/selector-path'
+import { OBSERVED_WINDOW_EVENTS } from '../../constants'
+import { AggregatedUserAction } from './aggregated-user-action'
+
+export class UserActionsAggregator {
+  /** @type {AggregatedUserAction=} */
+  #aggregationEvent = undefined
+  #aggregationKey = ''
+  #subscribers = []
+
+  /**
+   * @param {Function} storeMethod The function given by the parent class to call when an event is done aggregating
+   */
+  constructor () {
+    return this
+  }
+
+  #emit (event, data) {
+    this.#subscribers.forEach(subscriber => {
+      if (subscriber.event === event) {
+        subscriber.callback(data)
+      }
+    })
+  }
+
+  on (event, callback) {
+    this.#subscribers.push({ event, callback })
+  }
+
+  /**
+   * Process the event and determine if a new aggregation set should be made or if it should increment the current aggregation
+   * @param {Event} evt The event supplied by the addEventListener callback
+   */
+  process (evt) {
+    const selectorPath = getSelectorPath(evt)
+    const aggregationKey = getAggregationKey(evt, selectorPath)
+    if (aggregationKey === this.#aggregationKey) {
+      // an aggregation exists already, so lets just continue to increment
+      this.#aggregationEvent.aggregate(evt)
+    } else {
+      // store the prev existing one (if there is one)
+      this.storeCurrentUserActionInFeature()
```

Comment:
Return this to caller directly instead so feature agg can `.addEvent` on it... or pass it in `storeFn` if this is a helper method